recipe/deploy/prepare.php

```diff
@@ -36,4 +37,25 @@
 
     // Create shared dir.
     run("cd {{deploy_path}} && if [ ! -d shared ]; then mkdir shared; fi");
+
+    // Check and save the remote HEAD/revision and compare it with the existing saved one (if any)
+    // This avoid unnecessary releases when the last commit id matches the existing one (HEAD)
+    $repository  = trim(get('repository'));
+    $revision    = input()->getOption('revision') ?? null;
+    $remoteHead  = $revision ? $revision :
+        trim(run(sprintf('%s ls-remote %s HEAD | tr -d "HEAD"',
+            get('bin/git'), $repository)));
+    if (true === get('check_remote_head') && null == input()->getOption('tag')) {
+        $headPath = trim(get('deploy_path').'/.dep/HEAD');
+        $isRemoteHeadExists = test("[ -f $headPath ]");
```

Comment:
Is this check really required, or could it be achieved in combination with the „cat“ line in a single command?
Every remote command has a big overhead

Reply:
Well, `cat`ing non-existing file leads to `STDERR`.
I know, we could output the error to the `2>/dev/null` but I think it's a hack, the right solution is to check whether the file exists or not.

Reply:
And something along the lines of
`run('test -f '. $headPath ' && cat '.$headPath);`

Reply:
This is great.

Reply:
Nope, the https://en.wikipedia.org/wiki/Test_(Unix)